.circleci/validate_changed_seed.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/python3
+
+# Copyright (c) 2018 The Hyve B.V.
+# This code is licensed under the GNU Affero General Public License (AGPL),
+# version 3, or (at your option) any later version.
+#
+# This file is part of cBioPortal.
+#
+# cBioPortal is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""Validate changed seed - Validate if the new seed database does not contain any of the checked errors.
+"""
+
+import MySQLdb
+import sys
+import os
+import git
+import docker
+import urllib.request
+import time
+import _mysql_exceptions
+from time import gmtime, strftime
+
+
+def eprint(*args, **kwargs):
+    """Print error message and exit"""
+    print(*args, file=sys.stderr, **kwargs)
+    sys.exit(1)
+
+
+def check_changed_seed():
+    """Check whether the seed database files are changed"""
+
+    # Check whether the seed has changed
+    repo = git.Repo(os.path.dirname(os.getcwd()))
+    remote = repo.create_remote('upstream', url='https://github.com/cBioPortal/datahub/')
+    remote.fetch()
+    changed_files = repo.git.diff('upstream/master', name_only=True, diff_filter='ACMRU').split('\n')
+
+    seed_files = set()
+    for changed_file in changed_files:
+        if 'seedDB/seed-cbioportal_' in changed_file:
+            seed_files.add(changed_file)
+
+    if len(seed_files) == 0:
+        print('Seed database files are not changed')
+        sys.exit(0)
+    elif len(seed_files) > 1:
+        eprint('Found multiple seed databases. Unable to test whether seed is updated correctly.')
+
+    seed_file = list(seed_files)[0]
+    return seed_file
+
+
+def start_mysql_database(seed_location):
```

Comment:
@sandertan 	just curious why this part is scripted in Python? `docker-compose` file might be cleaner

Reply:
@inodb Thank you for the review. I use Python to check whether the seed files are changed and prepare the necessary files and variables to start a new seed database. I thought it's a nice flow if the seed database is started from the python code after these prerequisites. 

Also, the docker container is start from a docker container, but it runs as a sibling instead of as a child, because I use the `docker.sock` from the parent.